src/pages/index.js

```diff
@@ -0,0 +1,240 @@
+const editProfileNameInput = editProfileForm.elements["profile-name-input"];
+const editProfileDescriptionInput =
+  editProfileForm.elements["profile-description-input"];
+
+const newPostBtn = document.querySelector(".profile__add-btn");
+const newPostModal = document.querySelector("#new-post-modal");
+const addCardFormElement = document.forms["new-post-form"];
+const nameInput = newPostModal.querySelector("#card-caption-input");
+const linkInput = newPostModal.querySelector("#card-image-input");
+
+const profileNameEl = document.querySelector(".profile__name");
+const profileDescriptionEl = document.querySelector(".profile__description");
+const cardTemplate = document.querySelector("#card-template").content;
+const previewModal = document.querySelector("#preview-modal");
+const previewImage = previewModal.querySelector(".modal__image");
+const previewCaption = previewModal.querySelector(".modal__caption");
+
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+const avatarFormElement = document.querySelector("#edit-avatar-form");
+const avatarSubmitBtn = avatarModal.querySelector(".modal__submit-btn");
+const avatarModalCloseBtn = avatarModal.querySelector(".modal__close-btn");
+const avatarInput = avatarModal.querySelector("#profile-avatar-input");
+const avatarEditBtn = document.querySelector(".profile__avatar-btn");
+
+function getCardElement(data, currentUserId) {
+  const cardElement = cardTemplate.querySelector(".card").cloneNode(true);
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardTitleEl = cardElement.querySelector(".card__title");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  const cardLikeCount = cardElement.querySelector(".card__like-count");
+
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+  cardTitleEl.textContent = data.name;
+  cardLikeCount.textContent = Array.isArray(data.likes) ? data.likes.length : 0;
+
+  if (
+    Array.isArray(data.likes) &&
+    data.likes.some((like) => like._id === currentUserId)
+  ) {
+    cardLikeBtn.classList.add("card__like-btn_active");
+  }
+
+  cardImageEl.addEventListener("click", () => {
+    previewImage.src = data.link;
+    previewImage.alt = data.name;
+    previewCaption.textContent = data.name;
+    openModal(previewModal);
+  });
+
+  cardLikeBtn.addEventListener("click", () => {
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_active");
+    const method = isLiked ? api.unlikeCard : api.likeCard;
+
+    method
+      .call(api, data._id)
+      .then((updatedCard) => {
+        cardLikeBtn.classList.toggle("card__like-btn_active");
+        cardLikeCount.textContent = updatedCard.likes.length;
+      })
+      .catch(console.error);
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
```

Comment:
* The `confirmation` modal doesn't open when I click on the `trash` icon

You should open the modal here (not delete the card immediately)